src/main/java/io/pinecone/commons/IndexInterface.java

```diff
@@ -46,12 +86,43 @@ default UpsertRequest validateUpsertRequest(String id,
             vectorBuilder.setMetadata(metadata);
         }
 
-        upsertRequest.addVectors(vectorBuilder.build());
+        return vectorBuilder.build();
+    }
 
-        if (namespace != null) {
-            upsertRequest.setNamespace(namespace);
+    static VectorWithUnsignedIndices buildUpsertVectorWithUnsignedIndices(String id,
```

Comment:
Given the way java classes will work, if we want to support list<vectors> in upsert(), then the users have to build vectorsWithUnsignedIndices and exposing this method will help them with redundant vector building process.